runtime error

pods crash:
pod 1:
```
2021/05/27 10:00:14 listening on :8080
2021/05/27 10:01:07 failed to get Vector from http://10-42-2-4.adjacency.adjacency.svc.cluster.local:8080/vector?srv=_http._tcp.adjacency: failed to make GET request: Get "http://10-42-2-4.adjacency.adjacency.svc.cluster.local:8080/vector?srv=_http._tcp.adjacency": EOF
2021/05/27 10:01:08 failed to get Vector from http://10-42-1-105.adjacency.adjacency.svc.cluster.local:8080/vector?srv=_http._tcp.adjacency: failed to make GET request: Get "http://10-42-1-105.adjacency.adjacency.svc.cluster.local:8080/vector?srv=_http._tcp.adjacency": EOF
2021/05/27 10:01:10 failed to get Vector from http://10-42-2-4.adjacency.adjacency.svc.cluster.local:8080/vector?srv=_http._tcp.adjacency: failed to make GET request: Get "http://10-42-2-4.adjacency.adjacency.svc.cluster.local:8080/vector?srv=_http._tcp.adjacency": dial tcp 10.42.2.4:8080: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x39278c]

goroutine 1093 [running]:
main.timeHTTPRequest(0x502ef0, 0x4000190480, 0x400031acf0, 0x0)
	/adjacency/main.go:211 +0x21c
main.getLatencies.func1(0x400024b470, 0x502ef0, 0x4000190480, 0x400000ce58, 0x3, 0x3, 0x400000ceb8, 0x3, 0x3, 0x0)
	/adjacency/main.go:244 +0x74
created by main.getLatencies
	/adjacency/main.go:242 +0xd8
```
pod 2:
```
2021/05/27 09:59:55 listening on :8080
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x394b8c]

goroutine 757 [running]:
main.timeHTTPRequest(0x4bbc8c, 0x12f3ce0, 0x1416b90, 0x0)
	/adjacency/main.go:211 +0x20c
main.getLatencies.func1(0x1022b50, 0x4bbc8c, 0x12f3ce0, 0x100d000, 0x3, 0x3, 0x100d040, 0x3, 0x3, 0x2)
	/adjacency/main.go:244 +0x6c
created by main.getLatencies
	/adjacency/main.go:242 +0xc0
```
pod 3:
sorry can't get the logs

--- a/main.go
+++ b/main.go
@@ -208,15 +208,17 @@ func timeHTTPRequest(ctx context.Context, u *url.URL) *Latency {
 	var dur time.Duration
 	var res *http.Response
 	if res, err = httpPingClient.Do(req); err != nil {
-		defer res.Body.Close()
 		log.Printf("failed to make ping request: %v\n", err)
 		// set the time to almost infinity
 		dur = time.Duration(1<<63 - 1)
+	} else {
+		dur = time.Now().Sub(start)
 		if _, err := io.Copy(ioutil.Discard, res.Body); err != nil {
 			log.Printf("failed to discard body: %v\n", err)
 		}
-	} else {
-		dur = time.Now().Sub(start)
+		if err := res.Body.Close(); err != nil {
+			log.Printf("failed to close body: %v\n", err)
+		}
 	}
 	// Try to get IP address of target
 	// Shadow the err, because not being able to get an IP address should not

